[Cognitive Services] January Preview ARM API

fmabroukmsft opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [X] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.
- [X] A [release plan](https://aka.ms/azsdkdocs/release-plans) has been created. If not, please create one as it will help guide you through the REST API and SDK creation process. 

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.
- For guidance on SDK breaking change review, refer to https://aka.ms/ci-fix.

ramoka178 commented:
Please fix the [Swagger ModelValidation](https://github.com/Azure/azure-rest-api-specs/actions/runs/20146819016) and [Swagger LintDiff](https://github.com/Azure/azure-rest-api-specs/actions/runs/20146818984) errors

fmabroukmsft commented:
@dfuenzalida
@jvano

ramoka178 commented:
>         "summary": "List agents by subscription",

 and Resource Group

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2026-01-15-preview/cognitiveservices.json:7583 in a1f04d1. [](commit_id = a1f04d1bdeba7c57912724be86ee6a821f835e95, deletion_comment = False)

ramoka178 commented:
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CognitiveServices/agents/list": {

I dont see 'agents' being a resource type. 
How about this API route ending with 'providers/Microsoft.CognitiveServices/listAgents'

[https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/uri-format-and-arguments-for-crud-apis-on-resources.md](https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/uri-format-and-arguments-for-crud-apis-on-resources.md)

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2026-01-15-preview/cognitiveservices.json:7578 in a1f04d1. [](commit_id = a1f04d1bdeba7c57912724be86ee6a821f835e95, deletion_comment = False)

ramoka178 commented:
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CognitiveServices/agents/register": {

Microsoft.CognitiveServices/registerAgent?

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2026-01-15-preview/cognitiveservices.json:7717 in a1f04d1. [](commit_id = a1f04d1bdeba7c57912724be86ee6a821f835e95, deletion_comment = False)

ramoka178 commented:
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CognitiveServices/agents/unregister": {

Microsoft.CognitiveServices/unregisterAgent ?

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2026-01-15-preview/cognitiveservices.json:7782 in a1f04d1. [](commit_id = a1f04d1bdeba7c57912724be86ee6a821f835e95, deletion_comment = False)

ramoka178 commented:
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CognitiveServices/tools/validate": {

validateTools ?

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2026-01-15-preview/cognitiveservices.json:7977 in a1f04d1. [](commit_id = a1f04d1bdeba7c57912724be86ee6a821f835e95, deletion_comment = False)

ramoka178 commented:
>           },

Same as above.

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2026-01-15-preview/cognitiveservices.json:8027 in a1f04d1. [](commit_id = a1f04d1bdeba7c57912724be86ee6a821f835e95, deletion_comment = False)

ramoka178 commented:
>         "Desc"

use full wording

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2026-01-15-preview/cognitiveservices.json:14378 in a1f04d1. [](commit_id = a1f04d1bdeba7c57912724be86ee6a821f835e95, deletion_comment = False)

ramoka178 commented:
>         "id": {

'id' is typically used for top-level Response fields, like the azure resource id.

rename it appropriately.

Here and other places of this PR.

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2026-01-15-preview/cognitiveservices.json:14435 in a1f04d1. [](commit_id = a1f04d1bdeba7c57912724be86ee6a821f835e95, deletion_comment = False)

ramoka178 commented:
>       "additionalProperties": true

Use of additionalProperties is not allowed for properties owned by the service. The only time its ok to use it is when the properties are pass thru (user defined) and not subject to any validations. Please make this an array or provide an explanation for why you need this. 

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2026-01-15-preview/cognitiveservices.json:14568 in a1f04d1. [](commit_id = a1f04d1bdeba7c57912724be86ee6a821f835e95, deletion_comment = False)

ramoka178 commented on specification/cognitiveservices/resource-manager/readme.md:
```diff
@@ -30,7 +30,162 @@ These are the global settings for the CognitiveServices API.
+    reason: Approved to be suppressed in AML swagger.
+    where:
+      - $.definitions.ConnectionOAuth2.properties.clientId.format
+  - code: DeleteResponseCodes
```
For other comments that say to be complaint with existing behaviours.

we should consider this as an opportunity to fix things rather than carrying forward the existing issues.

ramoka178 commented on specification/cognitiveservices/resource-manager/readme.md:
```diff
@@ -30,7 +30,162 @@ These are the global settings for the CognitiveServices API.
+    reason: Same as existing account resource, trying to have the same behavior
+    where:
+      - $.definitions.ProjectProperties.properties.endpoints
+  - code: AvoidAdditionalProperties
```
>AvoidAdditionalProperties [](http://example.com/codeflow?start=10&length=25)

for all additionalProperties

The only case where these are approved is : if they are tags, if they are pass-through values that you are not doing any validation and are passing them to a different RP.

If you are handling these values, these properties should be schematized.

If you have a use-case to discuss more, please schedule an ARM modelling office hours and present your case.

fmabroukmsft replied:
The 'AvoidAdditionalProperties` suppressions here don't reflect any new changes in this api version so shouldn't be blocking. it's worth a discussion with the owners of this resource type though

fmabroukmsft commented:
> > ```
> >       },
> > ```
> 
> These can go in the ErrorResponse schema. You could remove these here.
> 
> Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2026-01-15-preview/cognitiveservices.json:7620 in [a1f04d1](https://github.com/Azure/azure-rest-api-specs/commit/a1f04d1bdeba7c57912724be86ee6a821f835e95). [](commit_id = [a1f04d1](https://github.com/Azure/azure-rest-api-specs/commit/a1f04d1bdeba7c57912724be86ee6a821f835e95), deletion_comment = False)

Removed in favor of default ErrorRespnose handling, thanks

fmabroukmsft commented:
> > ```
> >         "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
> > ```
> 
> v5/v6 , here and everywhere else in this PR.
> 
> Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2026-01-15-preview/cognitiveservices.json:7588 in [a1f04d1](https://github.com/Azure/azure-rest-api-specs/commit/a1f04d1bdeba7c57912724be86ee6a821f835e95). [](commit_id = [a1f04d1](https://github.com/Azure/azure-rest-api-specs/commit/a1f04d1bdeba7c57912724be86ee6a821f835e95), deletion_comment = False)

I attempted to move to v6 in the last version but was forced to downgrade back due to issues

fmabroukmsft commented:
> > ```
> >     "azureResource": {
> > ```
> 
> azureResourceId ?
> 
> Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/preview/2026-01-15-preview/cognitiveservices.json:14532 in [a1f04d1](https://github.com/Azure/azure-rest-api-specs/commit/a1f04d1bdeba7c57912724be86ee6a821f835e95). [](commit_id = [a1f04d1](https://github.com/Azure/azure-rest-api-specs/commit/a1f04d1bdeba7c57912724be86ee6a821f835e95), deletion_comment = False)

Fixed

razvanbadea-msft commented on specification/cognitiveservices/resource-manager/readme.md:
```diff
@@ -30,7 +30,126 @@ These are the global settings for the CognitiveServices API.
+input-file:
+  - Microsoft.CognitiveServices/preview/2026-01-15-preview/cognitiveservices.json
+suppressions:
+  - code:  ProvisioningStateMustBeReadOnly
```
>ProvisioningStateMustBeReadOnly [](http://example.com/codeflow?start=11&length=31)

is this needed? i see no changes on these paths

fmabroukmsft replied:
Removed, thanks

josefree commented:
All active Azure services must convert to TypeSpec by March 31, 2026. Starting July 1, 2026, SDK generation from the autorest toolchain will be blocked. Please reach out tspconversion@service.microsoft.com for details.

pshao25 reviewed (dismissed):
Please add new version based on TypeSpec.

fmabroukmsft committed:
Add 2025-12-01 stable API version for CognitiveServices Management

fmabroukmsft committed:
Add 2026-01-15-preview API version for CognitiveServices Management

fmabroukmsft committed:
Correct AgentApplication naming to AgenticApplication

fmabroukmsft committed:
Add newfoundryAutoUpgrade property to cog services swagger

fmabroukmsft committed:
Fix CI issues: example api-versions, readme tag, C# emitter config, suppression justifications

fmabroukmsft committed:
Fix CI issues for 2026-01-15-preview

- Fix api-version in 146 TypeSpec example files (2025-10-01-preview -> 2026-01-15-preview)
- Fix FIXMEs in AgenticApplication.tsp with proper justifications
- Add package-2026-01-15-preview tag with suppressions to readme.md
- Remove stale AgentApplication example directories from all output versions
- Regenerate swagger output

fmabroukmsft committed:
Fix api-version in 2025-12-01 AgenticApplication examples

The 7 AgenticApplication example files for 2025-12-01 had api-version
2025-10-01-preview instead of 2025-12-01. These were introduced by the
AgentApplication -> AgenticApplication rename commit and carried the
wrong version from the original preview examples.

fmabroukmsft committed:
Fix TypeSpec formatting in AgenticApplication.tsp and models.tsp

kazrael2119 committed:
Merge branch 'main' into fmabrouk/cogsvc-2025-12-01

fmabroukmsft committed:
Merge remote-tracking branch 'origin/fmabrouk/cogsvc-2025-12-01' into fmabrouk/cogsvc-2026-01-15-preview

Dchillakuru committed:
Adding delete managed network endpoint for FDP (#41136)

* adding delete endpoint for managed network along with example

* add delete related flow in type script files

fmabroukmsft committed:
Remove stale ProvisioningStateMustBeReadOnly suppressions and orphaned AgentApplication examples

fmabroukmsft committed:
Regenerate swagger from TypeSpec: add readOnly to provisioningState, ManagedNetwork delete operation and examples

fmabroukmsft committed:
Revert AgenticApplication rename, add 2026-01-15-preview models (QuotaScopeType, FoundryAutoUpgrade, scopeType/scopeId)

fmabroukmsft committed:
Version-gate ManagedNetwork delete to 2026-01-15-preview, fix operationId via @@clientLocation

fmabroukmsft committed:
Restore orphaned listDeployments.json to avoid Protected Files CI failure

fmabroukmsft committed:
Revert "Restore orphaned listDeployments.json to avoid Protected Files CI failure"

This reverts commit 21a1610f05bbaf051b59fa5324bfe375378c2689.

fmabroukmsft committed:
fix: correct api-version in AgentApplication examples from 2025-12-01 to 2026-01-15-preview

amitthakur-ms committed:
Add Global Quota Management support: GetUsages and Model Capacities APIs with scope type examples for 2026-01-15-preview

fmabroukmsft committed:
Merge main into fmabrouk/cogsvc-2026-01-15-preview

fmabroukmsft committed:
Add locations/computeOperations endpoint for 2026-01-15-preview

Dchillakuru committed:
fix managed vnet gaps for Jan preview version (#41498)

* fix swagger

* adding changes to tsp files

* add optional parameter for provision body

fmabroukmsft committed:
Add routing configuration for model router deployment

fmabroukmsft committed:
Fix example issues: typos, malformed GUIDs, invalid placeholders, and null values

fmabroukmsft committed:
Merge main into fmabrouk/cogsvc-2026-01-15-preview

fmabroukmsft committed:
Fix arm-post-response-codes crash and regenerate swagger

fmabroukmsft committed:
Fix ManagedNetwork examples: 201 -> 202 for async PUT operations

fmabroukmsft committed:
Revert unintended package-lock.json changes

fmabroukmsft committed:
Add missing managedNetworkName parameter to provisionManagedNetwork examples

fmabroukmsft committed:
Suppress NestedResourcesMustHaveListOperation for ComputeOperationStatus

fmabroukmsft committed:
Suppress arm-put-operation-response-codes linter errors

fmabroukmsft committed:
Fix tsp format

fmabroukmsft committed:
Fix incorrect resource ids in example files

Dchillakuru committed:
Fix managed network examples to include provisioning and rename egress public ip address (#41672)

* Fix managed network examples: rename egressIpAddress to firewallPublicIpAddress and add provisioningState inside managedNetwork settings

* Fix managed network examples: rename egressIpAddress to firewallPublicIpAddress and add provisioningState inside managedNetwork settings

* adding provisioningState in oct preview as well

* adding provisioningState for list network as well

fmabroukmsft committed:
Rename egressIpAddress to firewallPublicIpAddress and add provisioningState to ManagedNetworkSettings
